src/AggregateFunctions/AggregateFunctionUniq.h

```diff
@@ -182,9 +186,66 @@ struct OneAdder
                 StringRef value = column.getDataAt(row_num);
 
                 UInt128 key;
-                SipHash hash;
-                hash.update(value.data, value.size);
-                hash.get128(key.low, key.high);
+
+#if defined(__SSSE3__) && !defined(MEMORY_SANITIZER)
+                /// A trick for better performance: use last bit of key as a flag.
+                /// If string is not larger than 15 bytes, set the flag to zero and put the string itself into the key.
+                /// If it is larger - calculate it's cryptographic hash but set the last bit to one.
```

Comment:
As I can see from code, serialization of uniqExact state is a serialization of it 128 bit hashes.
If we add such optimization it will brake compatibility
* between old and current version
* between servers with and without SSSE3